lmcache/storage_backend/local_backend.py

```diff
@@ -563,3 +564,211 @@ def close(self):
 
     def __del__(self):
         self.close()
+
+
+@_lmcache_nvtx_annotate
+@torch.inference_mode()
+def save_disk_with_callback(
+    path: str,
+    kv_obj: KVObj,
+    key: CacheEngineKey,
+) -> Tuple[CacheEngineKey, float, int]:
+    save_file({"kv_chunk": kv_obj.data.contiguous()}, path)
+    return key, _get_size_in_gb(kv_obj.data), kv_obj.chunk_idx
+
+
+class LMCDiskBackend(LMCLocalDiskBackend):
+    """
+    Cache engine for storing the KV cache of the tokens in the local disk.
+    """
+
+    def __init__(self,
+                 config: LMCacheEngineConfig,
+                 metadata: LMCacheMemPoolMetadata,
+                 dst_device: str = "cuda"):
+        """
+        Throws:
+            RuntimeError if the loaded configuration does not match the current
+                configuration
+        """
+        super().__init__(config, metadata, dst_device)
+
+        assert config.disk_url is not None, (
+            "Need to specify local path if when "
+            "using LMCLocalDiskBackend")
+        self.proxy = xmlrpc.client.ServerProxy(config.disk_url)
+        # Info: Dict[str, Any] = self.proxy.Info()  # type: ignore
+        # self.remote_chunk_size = Info["chunk_size"]
+
+        self.cached_paths: Dict[CacheEngineKey, str] = {}
+
+        self.written_buffer_size: int = 10
+        self.remain_writing: int = 0
+
+        self.written_key: List[str] = []
+        self.written_size: List[float] = []
+
+        self.update_lock = threading.Lock()
+
+    def contains(
+        self,
+        key: CacheEngineKey,
+    ) -> bool:
+        anws: str = self.proxy.contains(key.to_string)  # type: ignore
+        self.cached_paths[key] = anws
+        return anws != ""
+
+    def batched_contains(self,
+                         keys: Iterable[CacheEngineKey],
+                         total_size: float = 0) -> Iterable[bool]:
+        if total_size == 0:
+            contains_paths: Iterable[str] = self.proxy.batched_contains(
+                [key.to_string() for key in keys])  # type: ignore
+            return [path != "" for path in contains_paths]
+        else:
+            # Write check
+            paths: Iterable[str] = self.proxy.batched_write_check(
+                [key.to_string() for key in keys], total_size)  # type: ignore
+
+            an = []
+            for path, key in zip(paths, keys):
+                if path == "":
+                    an.append(True)
+                else:
+                    self.cached_paths[key] = path
+                    an.append(False)
+            return an
+
+    def save_end_callback(self, future: Future):
```

Comment:
I'm a bit confused about the use of lock in this function? Why is the `batched_write_ready` dunction locked?

Reply:
We could talk about lock tomorrow. I'm not sure where I should use them.

Reply:
There is no lock here. You just inform the address manager that you have finished. You don't even update dict.